packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -669,4 +671,53 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('Two as an Arabic numeral: 2'), findsOneWidget);
   });
+
+  testWidgets('DropdownButton onTap callback is called when defined', (WidgetTester tester) async {
+    int dropdownButtonTapCounter = 0;
+    String value = 'one';
+    void didChangeValue(String newValue) {
+      value = newValue;
+    }
+
+    Widget build() => buildFormFrame(
+      value: value,
+      onChanged: didChangeValue,
+      onTap: () { dropdownButtonTapCounter += 1; },
+    );
+    await tester.pumpWidget(build());
+
+    expect(dropdownButtonTapCounter, equals(0));
+
+    // Tap dropdown button.
+    await tester.tap(find.text('one'));
+    await tester.pump();
```

Comment:
I think you just need to call `tester.pumpAndSettle()` without `tester.pump()`.
Do read the difference between them and why one makes the other obsolete.
Same as below.